proxy/proxy.go

```diff
@@ -20,33 +20,43 @@ type ContainerRequest struct {
 }
 
 type Proxy interface {
-	Serve() error
+	Serve(string, bool) error
 	Stop() error
 	GetRequests() ([]ContainerRequest, error)
 }
 
 type HTTPProxy struct {
-	address           string
-	containerRequests []ContainerRequest
-	listener          *net.TCPListener
-	errors            []error
+	address            string
+	containerRequests  []ContainerRequest
+	listener           *net.TCPListener
+	errors             []error
+	imageInspectsCount map[string]int
+	endpointHost       string
+	noBuild            bool
 }
 
 func NewHTTPProxy(listenAt string) *HTTPProxy {
 	return &HTTPProxy{address: listenAt}
 }
 
-func (p *HTTPProxy) Serve() error {
+func (p *HTTPProxy) Serve(endpointHost string, noBuild bool) error {
```

Comment:
this is kinda weak...
* The only place we need the endpointHost is in the build function. I couldn't really come up with an alternative though.
* The noBuild flag doesn't seem as out of place. Another thought I had was just passing in the action that is invoking this thing, and do the logic based on that knowledge, but I figured that was the responsibility of the action, hence the simple flag.